gulp/plugin.js

```diff
@@ -31,11 +30,25 @@ module.exports = function (gulp, gutil) {
   });
 
   gulp.task('plugin-build-res', function (cb) {
-    rsync()('res', 'plugin').then(function () { cb(); }, function () { handleRsyncError(cb); });
+    rsync()('res', 'plugin').catch(handleRsyncError).finally(cb);
   });
 
   gulp.task('plugin-build-web', function (cb) {
-    rsync()('web', 'plugin').then(function () { cb(); }, function () { handleRsyncError(cb); });
+    rsync()('web', 'plugin').catch(handleRsyncError).finally(cb);
+  });
+
+  gulp.task('plugin-copy-files', function (cb) {
+    if (Array.isArray(gutil.env.copyFiles)) {
+      var promises = [];
+      gutil.env.copyFiles.forEach(function (item) {
+        if (item.src && item.dest) {
+          promises.push(rsync()(item.src, item.dest));
+        }
+      });
+      Promise.all(promises).catch(handleRsyncError).finally(cb);
+    } else {
+      cb();
```

Comment:
is synchronous OK?

Reply:
Yes, synchronous would be OK, however gulp tasks are run async (unless a third party lib is used to circumvent).

See [here](https://github.com/gulpjs/gulp/blob/master/docs/API.md):

> Gulp tasks are asynchronous and Gulp uses async-done to wait for the task's completion. Tasks are called with a callback parameter to call to signal completion. Alternatively, Task can return a stream, a promise, a child process or a RxJS observable to signal the end of the task.

I could change it to return the Promise instead but seemed more readable this way to me.